source/tools/index.rst

```diff
@@ -68,6 +68,107 @@ This tool is available on GitHub: |https://github.com/intel/iasimage|
+
+    usage: GenContainer.py view [-h] -i IMAGE
+
+    optional arguments:
+      -h, --help  show this help message and exit
+      -i IMAGE    Container input image
+
+**create** command::
+
+    usage: GenContainer.py create [-h] (-l LAYOUT | -cl COMP_LIST [COMP_LIST ...])
+                                  [-t IMG_TYPE] [-o OUT_PATH] [-k KEY_PATH]
+                                  [-cd COMP_DIR] [-td TOOL_DIR]
+
+    optional arguments:
+      -h, --help            show this help message and exit
+      -l LAYOUT             Container layout intput file if no -cl
+      -cl COMP_LIST [COMP_LIST ...]
+                            List of each component files, following XXXX:FileName
+                            format
+      -t IMG_TYPE           Container Image Type : [NORMAL, CLASSIC, MULTIBOOT]
+      -o OUT_PATH           Container output directory/file
+      -k KEY_PATH           Input key directory/file
+      -cd COMP_DIR          Componet image input directory
+      -td TOOL_DIR          Compression tool directory
+
+**extract** command::
+
+    usage: GenContainer.py extract [-h] -i IMAGE [-n COMP_NAME] [-od OUT_DIR]
+                                  [-td TOOL_DIR]
+
+    optional arguments:
+      -h, --help    show this help message and exit
+      -i IMAGE      Container input image path
+      -n COMP_NAME  Component name to extract
+      -od OUT_DIR   Output directory
+      -td TOOL_DIR  Compression tool directory
+
+**replace** command::
+
+    usage: GenContainer.py replace [-h] -i IMAGE [-o NEW_NAME] -n COMP_NAME -f
+                                  COMP_FILE [-c {lz4,lzma,dummy}] [-k KEY_FILE]
+                                  [-od OUT_DIR] [-td TOOL_DIR]
+
+    optional arguments:
+      -h, --help           show this help message and exit
+      -i IMAGE             Container input image path
+      -o NEW_NAME          Container new output image name
+      -n COMP_NAME         Component name to replace
+      -f COMP_FILE         Component input file path
+      -c {lz4,lzma,dummy}  compression algorithm
+      -k KEY_FILE          Private key file path to sign component
+      -od OUT_DIR          Output directory
+      -td TOOL_DIR         Compression tool directory
+
+**sign** command::
+
+    usage: GenContainer.py sign [-h] -f COMP_FILE [-o SIGN_FILE]
+                                [-c {lz4,lzma,dummy}] [-a {rsa2048,sha256,none}]
+                                [-k KEY_FILE] [-od OUT_DIR] [-td TOOL_DIR]
+
+    optional arguments:
+      -h, --help            show this help message and exit
+      -f COMP_FILE          Component input file path
+      -o SIGN_FILE          Signed output image name
+      -c {lz4,lzma,dummy}   compression algorithm
+      -a {rsa2048,sha256,none}
+                            authentication algorithm
+      -k KEY_FILE           Private key file path to sign component
+      -od OUT_DIR           Output directory
+      -td TOOL_DIR          Compression tool directory
+

```

Comment:
can you provide some detailed command line example on how to create/replace/extract  a container image ?

Reply:
Ok.